packages/credentials/src/verifiable-credential.ts

```diff
@@ -250,20 +187,7 @@ export class VerifiableCredential {
    * ```
    */
   public static async verify(vcJwt: string): Promise<void> {
-    const jwt = decode(vcJwt); // Parse and validate JWT
-
-    // Ensure the presence of critical header elements `alg` and `kid`
-    if (!jwt.header.alg || !jwt.header.kid) {
-      throw new Error('Signature verification failed: Expected JWS header to contain alg and kid');
-    }
-
-    const verificationResponse = await verifyJWT(vcJwt, {
-      resolver: tbdResolver
-    });
-
-    if (!verificationResponse.verified) {
-      throw new Error('VC JWT could not be verified. Reason: ' + JSON.stringify(verificationResponse));
-    }
+    await CompactJwt.verify({ compactJwt: vcJwt });
```

Comment:
Note: this could be by our design, but this verify doesn't validate the payload.`validatePayload` could be thrown on the end of this if needed.
This could be a legit signed jwt but an invalid vcjwt.

Also here is the old verifyJwt lib - https://github.com/decentralized-identity/did-jwt/blob/master/src/JWT.ts#L486 it didn't really validate the payload either as far as I can tell but it did go into seeing if it was self issued and it looked at sub

Reply:
sry my response was confusing. we can totally validate payload inside `verify`. i mean in addition to that. like incase they need the subject or issuer did they don't have to go dig it out. or if they need to do something with payload after validation

Reply:
done! https://github.com/TBD54566975/web5-js/commit/483f5ee8ac5338423e4fef48791d165ddcabc4e8